tools/mo/openvino/tools/mo/utils/ir_engine/compare_graphs.py

```diff
@@ -116,6 +116,24 @@ def compare_graphs(graph: Graph, graph_ref: Graph, last_node: str, last_node_ref
                                       ''.format(node.id, cur_node_type, attr))
                         continue
 
+                    def align_strided_slice_masks(curr_node: Node, rank: int):
+                        for mask_name in StridedSlice.get_mask_names():
+                            if type(curr_node[mask_name]) == int:
+                                curr_node[mask_name] = [curr_node[mask_name]]
+                            elif type(curr_node[mask_name]) == str:  # if mask is an empty string ''
+                                assert len(curr_node[mask_name]) == 0
+                                curr_node[mask_name] = []
+
+                            num_insertions = rank - len(curr_node[mask_name])
+                            curr_node[mask_name] = np.append(curr_node[mask_name], [0] * num_insertions).astype(int)
+
+                    if cur_node_type == 'StridedSlice' and node.in_node(1).has('shape') \
+                            and node.in_node(1).shape is not None:
```

Comment:
Do we really need to align masks for rank? I was thinking that we can only have [0] if we have empty "" mask

Reply:
>Do we really need to align masks for rank?

We could've also aligned one mask to another, e.g. 'begin_mask' of `node` to the same mask of `node_ref`; but aligning into rank seemed as an already known practice, we already use such thing for SS https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/ops/strided_slice.py#L105-L111

>I was thinking that we can only have [0] if we have empty "" mask

Not necessary. Even if we have one empty mask "" other node could have `[0,0,0,0]` and they will behave the same. E.g. if slice rank is e.g. 4 and `begin_mask` for `node` is empty `""` and for `node_ref` is `[0,0,0,0]` they work exactly the same and they should be treated as equal during attrs comparison.